Wikipedia/Code/ActivityTabViewController.swift

```diff
@@ -0,0 +1,215 @@
+    var dataController: WMFActivityTabDataController
+    
+    public init(dataStore: MWKDataStore?, theme: Theme, viewModel: WMFActivityTabViewModel, dataController: WMFActivityTabDataController) {
+        self.dataStore = dataStore
+        self.viewModel = viewModel
+        let view = WMFActivityTabView(viewModel: viewModel)
+        self.hostingController = WMFActivityTabHostingController(rootView: view)
+        self.dataController = dataController
+        self.theme = theme
+        super.init()
+    }
+    
+    @MainActor required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Profile button dependencies
+
+    private var _yirCoordinator: YearInReviewCoordinator?
+    var yirCoordinator: YearInReviewCoordinator? {
+
+        guard let navigationController,
+              let yirDataController,
+              let dataStore else {
+            return nil
+        }
+
+        guard let existingYirCoordinator = _yirCoordinator else {
+            _yirCoordinator = YearInReviewCoordinator(navigationController: navigationController, theme: theme, dataStore: dataStore, dataController: yirDataController)
+            _yirCoordinator?.badgeDelegate = self
+            return _yirCoordinator
+        }
+
+        return existingYirCoordinator
+    }
+
+    private lazy var tabsCoordinator: TabsOverviewCoordinator? = { [weak self] in
+        guard let self, let nav = self.navigationController, let dataStore else { return nil }
+        return TabsOverviewCoordinator(
+            navigationController: nav,
+            theme: self.theme,
+            dataStore: dataStore
+        )
+    }()
+
+    private var _profileCoordinator: ProfileCoordinator?
+    private var profileCoordinator: ProfileCoordinator? {
+
+        guard let navigationController,
+              let yirCoordinator = self.yirCoordinator,
+              let dataStore else {
+            return nil
+        }
+
+        guard let existingProfileCoordinator = _profileCoordinator else {
+            _profileCoordinator = ProfileCoordinator(navigationController: navigationController, theme: theme, dataStore: dataStore, donateSouce: .historyProfile, logoutDelegate: self, sourcePage: ProfileCoordinatorSource.history, yirCoordinator: yirCoordinator)
+            _profileCoordinator?.badgeDelegate = self
+            return _profileCoordinator
+        }
+
+        return existingProfileCoordinator
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        Task {
+            if let (hours, minutes) = try? await dataController.getTimeReadPast7Days() {
+                viewModel.updateHoursMinutesRead(hours: hours, minutes: minutes)
+            }
+        }
+        
+        if let username = dataStore?.authenticationManager.authStatePermanentUsername {
+            viewModel.updateUsername(username: username)
```

Comment:
I like that you have the username as a @Published property. One thing to test for here is on this tab, if you are logged out, open up Profile > Log in. Upon successful login, does this username update and populate?